src/smart-components/request/requests.js

```diff
@@ -27,19 +28,31 @@ const Requests = ({ fetchRequests, requests, pagination, history }) => {
   };
 
   const routes = () => <Fragment>
-    <Route exact path="/requests/add_comment/:id" render={ props => <ActionModal { ...props } actionType={ 'Add Comment' } /> }/>
-    <Route exact path="/requests/approve/:id" render={ props => <ActionModal { ...props } actionType={ 'Approve' } /> } />
-    <Route exact path="/requests/deny/:id" render={ props => <ActionModal { ...props } actionType={ 'Deny' } /> } />
+    <Route exact path="/requests/add_comment/:id" render={ props => <ActionModal { ...props }
+      actionType={ 'Add Comment' }
+      preMethod = { fetchRequest }
+      postMethod={ fetchRequests } /> }/>
+    <Route exact path="/requests/approve/:id" render={ props => <ActionModal { ...props } actionType={ 'Approve' }
+      preMethod = { fetchRequest }
+      postMethod={ fetchRequests }/> } />
+    <Route exact path="/requests/deny/:id" render={ props => <ActionModal { ...props } actionType={ 'Deny' }
+      preMethod = { fetchRequest }
+      postMethod={ fetchRequests }/> } />
   </Fragment>;
 
-  const actionResolver = (requestData, { rowIndex }) => rowIndex === 1 ? null :
-    [
-      {
-        title: 'Comment',
-        onClick: () =>
-          history.push(`/requests/add_comment/${requestData.id}`)
-      }
-    ];
+  const actionResolver = (requestData, { rowIndex }) => {
+    return (rowIndex === 1 || areActionsDisabled(requestData) ? null :
```

Comment:
this looks like an extra `return` statement too

Reply:
I think the return is required here.